CMakeLists.txt

```diff
@@ -0,0 +1,40 @@
+cmake_minimum_required (VERSION 3.12)
+project (tn5250 LANGUAGES C VERSION 0.9.8)
+
+include(CheckIncludeFile)
+
+find_package(OpenSSL)
+find_package(Curses)
+
+check_include_file("dlfcn.h" HAVE_DLFCN_H)
+check_include_file("fcntl.h" HAVE_FCNTL_H)
+check_include_file("inttypes.h" HAVE_INTTYPES_H)
+check_include_file("locale.h" HAVE_LOCALE_H)
+check_include_file("pwd.h" HAVE_PWD_H)
+check_include_file("stdint.h" HAVE_STDINT_H)
+check_include_file("stdio.h" HAVE_STDIO_H)
+check_include_file("stdlib.h" HAVE_STDLIB_H)
+check_include_file("strings.h" HAVE_STRINGS_H)
+check_include_file("string.h" HAVE_STRING_H)
+check_include_file("syslog.h" HAVE_SYSLOG_H)
+check_include_file("sys/stat.h" HAVE_SYS_STAT_H)
+check_include_file("sys/time.h" HAVE_SYS_TIME_H)
+check_include_file("sys/types.h" HAVE_SYS_TYPES_H)
+check_include_file("sys/wait.h" HAVE_SYS_WAIT_H)
+check_include_file("unistd.h" HAVE_UNISTD_H)
+
+configure_file(config-cmake.h.in config.h @ONLY)
+
+if (${CMAKE_HOST_SYSTEM_NAME} MATCHES "Windows")
+    add_definitions( -D__WIN32__)
```

Comment:
We should probably change the source to use the standard `_WIN32` instead.

Reply:
I've also defined `_WIN32` but left the existing `__WIN32__` as there is code that already depends on that - it won't build if only `_Win32` is defined. I don't want to destabilize the existing build system.

Reply:
Sorry, I should have been more clear. What I meant was to modify the source files that use `__WIN32__` to use `_WIN32` instead so it builds without this extra definition. Windows compilers already predefine `_WIN32` so there's no point in defining it ourselves. Also, the existing Windows system is dead and I was going to remove it anyway.

I can make the `_WIN32` changes though and clean that up after if you just want me to merge as-is, though.